src/back/glsl/mod.rs

```diff
@@ -1341,11 +1341,21 @@ impl<'a, W: Write> Writer<'a, W> {
                 //TODO: handle MS
 
                 //Write the function to be used depending on the sample level
-                let fun_name = match level {
+                let mut fun_name = match level {
                     crate::SampleLevel::Auto | crate::SampleLevel::Bias(_) => "texture",
                     crate::SampleLevel::Zero | crate::SampleLevel::Exact(_) => "textureLod",
                     crate::SampleLevel::Gradient { .. } => "textureGrad",
                 };
+
+                // textureLod on sampler2DArrayShadow and samplerCubeShadow does not exist in GLSL.
+                // To emulate this, we will have to use textureGrad with a constant gradient of 0.
+                let workaround_lod_array_shadow_as_grad =
```

Comment:
This should only be active when we actually want to use the `textureLod`. Currently in this code, it may end up replacing `texture` with `textureGrad` for no reason.

Reply:
It would be even better if we don't compare strings here. Strings are the product of the backend, it shouldn't feed them it its own logic, which operates on the stronger IR. So instead of comparing to string, this should do:
```rust
let workaround_lod_array_shadow_as_grad = array_index.is_some() && depth_ref.is_some();
              let mut fun_name = match level {
                  crate::SampleLevel::Auto | crate::SampleLevel::Bias(_) => "texture",
                  crate::SampleLevel::Zero | crate::SampleLevel::Exact(_) => {
if workaround_lod_array_shadow_as_grad { "textureGrad" } else { "textureLod" },
                  crate::SampleLevel::Gradient { .. } => "textureGrad",
              };
```
And then check for `workaround_lod_array_shadow_as_grad` in the another match we do